frlearn/uncategorised/fuzzy_quantifiers.py

```diff
@@ -0,0 +1,104 @@
+"""
+    Some fuzzy quantifiers.
+    All the following functions are the implementations of defintions of some fuzzy quantifiers in [1].
+
+    References
+    ----------
+    [1] THEERENS, Adnan et CORNELIS, Chris.
+    Fuzzy rough sets based on fuzzy quantification.
+    Fuzzy Sets and Systems, 2023, vol. 473, p. 108704.
+"""
+
+
+def choquet_integral(f, mu):
+    """
+        Implementation of the Choquet_integral
+        parameters:
+        ----------
+        f: list
+           Define a list that contains the values of the function
+        mu: list
+            Define a list that contains the values of the monotone measure.
+    """
+    n = len(mu)
+    f_sorted = sorted(f)
```

Comment:
I am wondering whether this implementation makes sense. If I understand it correctly, the way it is used here, `mu` should contain the values of the monotone measure applied to A*_i as defined in Definition 2.9. But A*_i depends on the values of f.

So if we assume that `mu` is correctly specified in this way, shouldn't we also assume that `f` is also correctly specified, i.e. that it is already sorted?

But more generally, and more importantly, is there any use case for which we want to calculate a choquet integral directly? If yes, that should guide the way that we define this function ­— if no, there's no point including this function here. 

At the moment, it seems a bit unrealistic that a user can specifically supply the values of the measure applied to A*_i.

@amtheere can you weigh in here?

Reply:
It is necessary to sort the values of the functions to be integrated in ascending order to avoid negative results, for example. This is how the Choquet's integral is defined. 

We have defined the Choquet integral in the most general way possible, to avoid using the base set. But in practice, you need to know the values of this set to get the right results.

In order to avoid using the values of the basis set, I've had to dispense with the definition of the measurement function in the various quantifiers and have done so instead as an example (which isn't very realistic); this is why you get the impression that these quantifiers (binary Yaguer's model and binary WOWA's model) are the same; they differ fundamentally in the definition of the measurement function.

Finally, this function that compute the Choquet integral is never used explicitly by a user. We thought we'd move it to the "uncategorized" folder and then use it to evaluate the various quantifiers

Reply:
General comment: Everything I've done so far has been to avoid using the base set as a parameter for the quantifiers; but obviously this is absolutely necessary for everything to be correct.